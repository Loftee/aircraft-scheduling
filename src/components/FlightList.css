.flight-list-wrapper {
    width:37.5%;
    padding: 15px;
    float:left;
    list-style:none;
    margin:0;
}

.flight-list{
    width:100%;
    float:left;
    list-style:none;
    padding:0;
}

.flight-list-item {
    width:100%;
    margin-bottom: 20px;
    padding:10px;
    border: 1px solid #ccc;
    cursor: pointer;
}

.flight-list-item:hover {
    background: #eee;
    border-color: #999;
}

.flight-list-item__data {
    font-size: 0;
}

.flight-list-item__data span {
    font-size: initial;
    display: inline-block;
    width: 40%;
    vertical-align: middle;
    height: 20px;
}

.flight-list-item__data span:last-child {
    text-align: right;
}

span.flight-list-item__data-arrow {
    width: 20%;
    text-align: center;
    background: url('../assets/arrow.png') center center no-repeat;
    background-size: contain;

}

.flight-list-reset {
    float: left;
    width:100%;
    text-align:right;
}