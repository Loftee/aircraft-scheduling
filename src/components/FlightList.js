import React, {Component} from 'react';

import './FlightList.css';

class FlightList extends Component {
    render() {
        const {filteredFlightsList, selectFlight, resetFlights} = this.props;
        return (
            <div className={'flight-list-wrapper'}>
                <div className={"flight-list-reset"}>
                    <button className={'flight-list-reset-button'} onClick={() => {resetFlights()}}>reset</button>
                </div>
                <ul className={'flight-list'}>
                    {
                        filteredFlightsList.map((flight, index) => {
                            return (
                                <li
                                    key={index}
                                    className={'flight-list-item'}
                                    onClick={() => selectFlight(flight.id)}
                                >
                                    <h3 className={'flight-list-item__header'}>Flight: {flight.id}</h3>
                                    <div className={'flight-list-item__data'}>
                                        <span>Departure: {flight.readableDeparture} - {flight.origin}</span>
                                        <span className={'flight-list-item__data-arrow'} />
                                        <span>Arrival: {flight.readableArrival} - {flight.destination}</span>
                                    </div>
                                </li>
                            )
                        })
                    }
                </ul>
            </div>
        )
    }
}

export default FlightList;