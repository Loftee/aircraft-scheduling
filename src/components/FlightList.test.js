import React from 'react';
import {shallow} from 'enzyme';
import FlightList from './FlightList';

describe('Flight List', () => {
    const id = "AS1001";
    const props = {
        filteredFlightsList:
            [{
                id: id,
                departureTime: 21600,
                arrivalTime: 26100,
                readableDeparture: "06:00",
                readableArrival: "07:15",
                origin: "LFSB",
                destination: "LFMN"
            }]
    };

    it('renders correctly', () => {
        const component = shallow(<FlightList {...props} />);
        expect(component).toMatchSnapshot();
    });

    describe('when rendered', () => {
        const component = shallow(<FlightList {...props} />);

        it('returns a list of flights', () => {
            expect(component.find('.flight-list-item').length).toEqual(1);
        });

        describe('and a flight is clicked', () => {
            const mockSelect = jest.fn();
            const component = shallow(<FlightList {...props} selectFlight={mockSelect}/>);
            beforeEach(() => {
                component.find('.flight-list-item').simulate('click');
            });

            it('sends correct id', () => {
                expect(mockSelect).toHaveBeenCalledWith(id);
            });
        });

        describe('and a reset is clicked', () => {
            const mockReset = jest.fn();
            const component = shallow(<FlightList {...props} resetFlights={mockReset}/>);
            beforeEach(() => {
                component.find('.flight-list-reset-button').simulate('click');
            });

            it('sends correct id', () => {
                expect(mockReset).toHaveBeenCalled();
            });
        });
    });
});