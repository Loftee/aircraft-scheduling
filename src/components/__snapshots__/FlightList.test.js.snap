// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight List renders correctly 1`] = `
<div
  className="flight-list-wrapper"
>
  <div
    className="flight-list-reset"
  >
    <button
      className="flight-list-reset-button"
      onClick={[Function]}
    >
      reset
    </button>
  </div>
  <ul
    className="flight-list"
  >
    <li
      className="flight-list-item"
      key="0"
      onClick={[Function]}
    >
      <h3
        className="flight-list-item__header"
      >
        Flight: 
        AS1001
      </h3>
      <div
        className="flight-list-item__data"
      >
        <span>
          Departure: 
          06:00
           - 
          LFSB
        </span>
        <span
          className="flight-list-item__data-arrow"
        />
        <span>
          Arrival: 
          07:15
           - 
          LFMN
        </span>
      </div>
    </li>
  </ul>
</div>
`;
